src/components/astra-editor/index.ts

```diff
@@ -0,0 +1,351 @@
+        }  
+      `
+    }
+
+    const styleElement = document.createElement('style')
+    styleElement.innerHTML = `
+      .cm-tooltip-autocomplete ul::-webkit-scrollbar,
+      .cm-scroller::-webkit-scrollbar {
+        width: 10px;
+        height: 10px;
+      }
+
+      .cm-scroller::-webkit-scrollbar-thumb {
+        border-radius: 5px;
+      }
+
+      .cm-tooltip-autocomplete ul::-webkit-scrollbar-thumb,
+      .cm-scroller::-webkit-scrollbar-thumb {
+        background: rgba(0, 0, 0, 0.3);
+      }
+
+      .dark .cm-tooltip-autocomplete ul::-webkit-scrollbar-thumb, 
+      .dark .cm-scroller::-webkit-scrollbar-thumb {
+        background: rgba(255, 255, 255, 0.3);
+      }
+
+      .cm-tooltip-autocomplete > ul > li {
+        display: flex;
+      }
+
+      .cm-tooltip-autocomplete > ul > li > .cm-completionIcon {
+        width: 1em !important;
+        display: flex;
+        align-self: center;
+        justify-content: center;
+      }
+
+      .cm-tooltip-autocomplete .cm-completionLabel {
+        flex-grow: 1;
+      }
+
+      .cm-tooltip-autocomplete .cm-completionDetail {
+        padding-left: 15px;
+      }
+      
+      .cm-completionIcon-property::after {
+        content: "\\61" !important;
+        font-family: 'outerbase-icon' !important;
+      }
+
+      .cm-completionIcon-type::after {
+        content: "\\62" !important;
+        font-family: 'outerbase-icon' !important;
+      }
+
+      .cm-completionIcon-function::after,
+      .cm-completionIcon-method::after,
+      .cm-completionIcon-variable::after,
+      .cm-completionIcon-namespace::after,
+      .cm-completionIcon-interface::after {
+        content: '⚡' !important;
+      }
+    `
+
+    this.shadowRoot?.append(styleElement)
+
+    this.container = container
+  }
+
+  connectedCallback(): void {
+    this._color = this.getAttribute('color') === 'dark' ? 'dark' : 'light'
+    this._theme = this.getAttribute('theme') ?? 'moondust'
+    this._placeholder = this.getAttribute('placeholder') ?? ''
+
+    if (this._color === 'dark') {
+      this.container.classList.add('dark')
+    } else {
+      this.container.classList.remove('dark')
+    }
```

Comment:
we probably want this in `willUpdate` or `updated` cycle hooks, otherwise if it's changed at runtime we won't add/remove that class

after reading more of the code though, I see you're using the setters to maintain this -- we might find the code to be more succinct if we just catch it in the `updated()` hook and can remove all the custom setter/getter logic

https://lit.dev/docs/components/lifecycle/#reactive-update-cycle

Reply:
Let me explore this option.